wagtail/models/pages.py

```diff
@@ -2317,6 +2317,16 @@ def can_move_to(self, destination):
         ):
             return False
 
+        # if the page has a max_count_per_parent, check for a page of this type under the destination
+        if self.page.specific.max_count_per_parent is not None:
```

Comment:
```suggestion
        if self.page.specific_deferred.max_count_per_parent is not None:
```

as we don't need to full specific page object